internal/generator/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"errors"
+	"log"
+	"os"
+	"strings"
+)
+
+var errUnmachedBraces = errors.New("braces are unmatched")
+
+func main() {
+	args := os.Args
+
+	if len(args) != 2 {
+		log.Fatalln("wrong amount of arguments")
+	}
+
+	if len(args[1]) == 0 {
+		log.Fatalln("given regex is empty")
+	}
+
+	regex := args[1]
+
+	result, err := generateCode(regex)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Stdout.WriteString(result + "\n")
+}
+
```

Comment:
```suggestion
```

I think here duplicate of main.

We need only GenerateCode in this file, and use it in cmd/generator/main.go.

_We needed to move `generateCode` to a separate package in order to simplify cmd and reuse this function in the future._

Reply:
Don't remove, but replace actual main in cmd/generator